app/controllers/users_controller.rb

```diff
@@ -0,0 +1,79 @@
+class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :load_user, only: [:show, :edit, :destroy]
+
+  def current_user? user
+    user == current_user
+  end
+
+  def show
+    flash[:warning] = t "not_found" if @user.nil?
```

Comment:
điều hướng về page khác, đừng render về show nữa nhé
```suggestion
    unless @user
      flash[:warning] = t "not_found"
      redirect_to ....path
    end
```